### DISCUSSION TITLE
Atmospheric effects

### DISCUSSION THREAD

**Discussion by mefistotelis** [swfans/syndwarsfx#119] [General]
The game has rain, but there also seem to be snow - try adding:

```
diff --git a/src/game.c b/src/game.c
index 8ed1b98..8a41cbc 100644
--- a/src/game.c
+++ b/src/game.c
@@ -3755,6 +3755,7 @@ void draw_screen(void)
     struct DrawItem *itm;
     short n;
     ushort i;
+gamep_scene_effect = 2;

     p_bucket = &buckets[9999];
     if (dword_19F4F8)

```
Though I understand why they did not used it - looks more like noise.
And even if the effect was better, there are no snowy textures.
And would snowy textures even make sense in a cyberpunk game?

**Comment by Moburma** [comment]
I haven't tried the above yet so am posting in ignorance (as usual!) but I would suggest one of two things:

1. Maybe it was to be a ["TV snow"](https://www.youtube.com/watch?v=hSGNXY_CEbg) effect, to simulate some kind of interference? That definitely has prescedence in cyberpunk, as per the beginning of Neuromancer
2. A bit more esoteric, something I meant to write up ages ago in the Unguided campaign thread: The culmination of Ko Paull Vissick's plan in the game's plot is to send the unguided and remaining free people to Tibet. Specifically to a plateau in Tibet - the Tibetan Plateau is the highest and largest plateau in the world and includes Mount Everest. Presumably the zealot death ray thing would not harm people at such altitudes, but more relevantly there would likely be a a lot of snow here. Maybe it was for that?

**Comment by mefistotelis** [comment]
So it was supposed to be a death ray rather than ionosphere stripper?

I guess the static effect is possible, though the dots are a bit too sparse. And there is no intensity control.
Maybe it was a hailstorm rather than snow? For snow, I'm pretty sure Bullfrog team would know the flakes need to be slow moving..

There's also no intensity control in rain - it would be more interesting if the effect would strengthen over time..

![synII080](https://github.com/swfans/swars/assets/11463396/d27c0be9-f536-4bfe-b3f8-ff37575fbe10)
![synII081](https://github.com/swfans/swars/assets/11463396/3d6762ab-ee19-425f-990f-7d9561fbcf98)

EDIT:
And yes - it works properly only in low res.

**Reply by Moburma** [reply]
> So it was supposed to be a death ray rather than ionosphere stripper?

No, I was being tongue in cheek, ultimately the storyline is just the old pulp sci-fi trope of the baddies destroying the world with a "death ray". But clearly there was meant to be something special about Tibetan plateaus that would somehow mitigate this, and what is special about the area in real life is its altitude, so I assume it must have been to do with that.

**Reply by Moburma** [reply]
I just realised something - the climax of the plot is literally the same as that of [MoonRaker](https://www.imdb.com/title/tt0079574/?ref_=tt_urv)!
i.e. baddies kill off the world's population to replace it with beings in their own image and retreat to space/the Moon to do so.
Made me laugh when the penny dropped.

**Comment by Moburma** [comment]
I finally got round to playing with this, for me I think it's a "TV snow" effect, maybe some kind of thing for interference? Like they'd be some weapon used that would cause it for a while or something.

**Reply by mefistotelis** [reply]
I just rewrote the snow function.
It puts less of the snow at bottom of the screen, and uses XZ angle during the placement (though not properly, it uses the angle as linear value).

So I'm pretty sure it was supposed to be snow/hail, just not really finished.

EDIT: It was snow. After switching to proper random number generator, it started looking like snow.
I also has to introduce proper angle computations for the snow to work properly while moving view.

And btw, there's a code which modifies textures while it's snowing. It takes a few minutes for it to take effect, but then you start seeing how the snow affects textures (mostly looks fine, but any stretched textures look weird. And only first texture is altered, so any textures from the rest of the atlas do not match..

**Comment by mefistotelis** [comment]
Should we include atmospheric effects as an option in missions config?

The current solution (selection based on how many things are on the level) doesn't seem the best. 
First, we will continue modifying levels, and we do not necessarily want to modify effects as well.
Second, having that in config would allow to also set intensity of the rain, or use the snow (which is now working far better than before).

Maybe there should be snowing during Colombo mission, to indicate weather control system is abused?

And we can use light rain in some levels?

The hack to enable snow in current sunny levels:
```
diff --git a/src/game.c b/src/game.c
index 66e9044..041b7ac 100644
--- a/src/game.c
+++ b/src/game.c
@@ -4788,7 +4788,7 @@ void init_level(void)
     else if ((things_used & 3) || (current_map == 30) || (in_network_game)) // map030 london
     {
         LbFileLoadAt("data/tex00.dat", vec_tmap[0]);
-        gamep_scene_effect_type = ScEff_NONE;
+        gamep_scene_effect_type = ScEff_SNOW;
         gamep_scene_effect_intensity = 1000;
     }
     else
```

**Reply by Moburma** [reply]
I'm definitely in favour of making weather a mission config option. As you say it's too finicky once you start making new levels or changing existing ones. It seems it was likely a fairly late game addition so it probably made sense to do it how they did, but having actual control on how a level looks seems a  no-brainer improvement.

I just tried the snow effect, it's great! Definitely be good to have that as an option in levels. Is there any end to the unused stuff in this game?

**Reply by mefistotelis** [reply]
Ha, definitely another good feature to show on your YT channel.

There is probably still some stuff to unveil, but a lot more stuff could be found in materials from Bullfrog, if we had any.
I'm pretty sure all the sprites were rendered in higher res. These would make considerable impact on the game look if added.
Also the editors - we have some chunks of them in pre-alpha, but that's old version and so fragmented that it's hard to tell which parts match.

I wonder if there was an early implementation of the sonic blast shot - effects of bending the rendering area (called lense effects) were used in DK (ie. when possessing a Fly or Bug), I'd imagine the Sonic Blast would do something similar.

**Reply by Moburma** [reply]
Haha, was one step ahead of you making a new video!

There were clearly sprites for a female shady guy and a couple of other characters according to the pre-alpha, shame they didn't include them with the demo. Sonic Blast is a good call, there is comparatively a lot of stuff for that left in the game's files. I'm actually surprised there is no weapon select sample for it.
